script/hooks/migration.js

```diff
@@ -103,6 +103,20 @@ const migrateItemData = (item, worldSchemaVersion) => {
             update["data.spellType"] = "SPELL.SPELL";
         }
     }
+    if(worldSchemaVersion < 4.1) {
+        if (item.type === "weapon" && typeof item.data.features === "string") {
+            // Change features from string to object
+            update["data.features"] = {
+                "edged": false,
+                "pointed": false,
+                "blunt": false,
+                "parrying": false,
+                "hook": false,
+                "slowReload": false,
+                "others": item.data.features
+            }
```

Comment:
Since we are doing migration, I think we could one-up the effort here and look for features players may have input already.
```js
			// Change features from string to object
			const features = item.data.features
				.replace(".", "")
				.replace("loading is a slow action", "slowReload")
				.replace("slow reload", "slowReload")
				.split(", ");
			update["data.features"] = {
				edged: false,
				pointed: false,
				blunt: false,
				parrying: false,
				hook: false,
				slowReload: false,
				others: "",
			};
			let otherFeatures = "";
			for (const feature of features) {
				const lcFeature = feature === "slowReload" ? feature : feature.toLowerCase();
				if (lcFeature in update.data.features) {
					update.data.features[lcFeature] = true;
				} else {
					otherFeatures += feature + ", ";
				}
			}
			update.data.features.others = otherFeatures.substr(
				0,
				otherFeatures.length - 2
			);
```
This solution obviously works only with English. A bit more cumbersome but I think somethign could be worked out that is localization friendly.

Reply:
I initially decided _not_ to do this, because I though it would not work too well in most situations.
BUT… since Fria Ligan just released the core module and campaign contents, it would be good to make sure _at least_ those get migrated properly.

I don't own those: can you tell if features are only english strings or do they support multiple languages? We could make a special case just for it. What do you think?

Reply:
That is certainly an option. (And tbh I would have patched it in regardless to save me and a lot of people work). 

The solution above should work with the official compendia. I'll do a test tonight. I wrote this thing at work and honestly haven't been able to check if it works..